modules/desktop/graphics/ewwbar.nix

```diff
@@ -307,43 +319,37 @@ let
     runtimeInputs = [
       pkgs.gawk
       pkgs.pulseaudio
-      audio-ctrl
+      pkgs.pamixer
     ];
     bashOptions = [ ];
     text = ''
-        popup_timer_pid=0
-
-        show_popup() {
-          if ! ${eww} active-windows | grep -q "quick-settings"; then
-              # Reset timer
-              if [ "$popup_timer_pid" -ne 0 ]; then
-                kill "$popup_timer_pid" 2>/dev/null
-                popup_timer_pid=0
-              fi
-
-              if ! ${eww} active-windows | grep -q "volume-popup"; then
-                ${eww} open volume-popup
-                ${eww} update volume-popup-visible="true"
-              fi
-              (
-                sleep 2
-                ${eww} update volume-popup-visible="false"
-                sleep 0.1
-                ${eww} close volume-popup
-              ) &
-
-              popup_timer_pid=$!
-          fi
-        }
+      export PULSE_SERVER=audio-vm:4713
+      popup_timer_pid=0
 
-      volume_level() {
-          audio-ctrl get | awk '{print $1 + 0.0}'
-      }
+      show_popup() {
+        if ! ${eww} active-windows | grep -q "quick-settings"; then
+            # Reset timer
+            if [ "$popup_timer_pid" -ne 0 ]; then
+              kill "$popup_timer_pid" 2>/dev/null
+              popup_timer_pid=0
+            fi
 
-      is_muted() {
-          audio-ctrl get_mut
+            if ! ${eww} active-windows | grep -q "volume-popup"; then
+              ${eww} open volume-popup
+              ${eww} update volume-popup-visible="true"
+            fi
+            (
+              sleep 2
```

Comment:
Is this really the way to get popups dissapear after timer in eww?
If it is it seems re-usable code to me?

Reply:
eww does provide functionality to open a window with a timer **`eww open --duration [time]`**, but no way to reset or cancel the timer once the window is open.
This resulted in jittery behavior, where a user might keep increasing the brightness a few times in a row but the popup would suddenly disappear because the timer runs out and then appear immediately again.

It is reusable code, however tracking the PID of the current timer for that specific window would require storing the PID somewhere externally, which I wanted to avoid for the sake of simplicity and performance.